spring-xd-spark-streaming-tests/src/test/java/org/springframework/xd/spark/streaming/AbstractSparkStreamingTests.java

```diff
@@ -108,12 +110,65 @@ public void testSparkProcessor() throws Exception {
 	}
 
 	@Test
+	public void testSparkProcessorWithInputType() throws Exception {
+		HttpSource source = new HttpSource(shell);
+		String streamName =  testName.getMethodName()  + new Random().nextInt();
+		FileSink sink = new FileSink().binary(true);
+		try {
+			String stream = String.format("%s | spark-word-count --inputType=text/plain | %s " +
+					"--inputType=text/plain", source, sink);
+			createStream(streamName, stream);
+			source.ensureReady().postData(TEST_MESSAGE);
+			assertThat(sink, XDMatchers.eventually(XDMatchers.hasContentsThat(equalTo("(foo,3)"))));
+		}
+		finally {
+			streamOps.destroyStream(streamName);
+			sink.cleanup();
+		}
+	}
+
+	@Test
+	public void testSparkProcessorWithOutputType() throws Exception {
+		HttpSource source = new HttpSource(shell);
+		String streamName =  testName.getMethodName()  + new Random().nextInt();
+		FileSink sink = new FileSink().binary(true);
+		try {
+			String stream = String.format("%s | spark-word-count --outputType=application/json | %s", source, sink);
+			createStream(streamName, stream);
+			source.ensureReady().postData(TEST_MESSAGE);
+			assertThat(sink, XDMatchers.eventually(XDMatchers.hasContentsThat(equalTo("{\"_1\":\"foo\",\"_2\":3}"))));
+		}
+		finally {
+			streamOps.destroyStream(streamName);
+			sink.cleanup();
+		}
+	}
+
```

Comment:
Agreed, although having at least ONE test of each kind makes sure that the rest of the infrastructure works correctly in a real setting..